### ISSUE TITLE
v4 Release

### ISSUE THREAD

**Issue by zvirja** [AutoFixture/AutoFixture#866]
Here is the list of actions required to prepare for the v4 release.

Preparations:
- [x] Close all the PRs for `master` and freeze changes to the `master` branch till the release.
- [x] Close all the issues and PRs in the [v4 milestone](https://github.com/AutoFixture/AutoFixture/milestone/3).
- [x] Review all the v4 changes and compose the breaking changes document on Wiki (#888).
- [x] Update ReadMe with link to the breaking changes.
- [x] Review `Contributing.md` and wiki and ensure no obsolete information is present there.

Release:
- [x] Release AutoFixture `v4.0.0-rc1`.
- [x] Release the SemanticComparison library as it's located in the standalone repo.
- [x] Release AutoFixture and make a tweet.
- [x] Thank to everybody who participated 😉

**Comment by zvirja** [comment]
I have an idea to release a RC before releasing the final version. It could happen that something is broken, so we could get a feedback and fix that. Everything will be ready till that moment, so users could start to upgrade. If there are some last-minute issues, we could fix them before the final release. If not - we'll simply proceed it to RTM.

My concern is that this release contain too many changes - from broken changes to new HUGE features (like .NET Core support, reworked NSubstitute integration approach). It could easily happen that we overlooked something and RC would work fine here.

@AutoFixture/core What do you think?

**Comment by moodmosaic** [comment]
Do `RC` and `RTM` play well with SemVer? I personally *hate* these `RT` and `RTM` versioning schemes 😆 If it's ready to release, then maybe it should be released (?) If something's broken, well, it'll be fixed (?)

**Comment by zvirja** [comment]
> Do RC and RTM play well with SemVer? I

Sure, it's [described here](http://semver.org/#spec-item-9).

> If it's ready to release, then maybe it should be released (?) If something's broken, well, it'll be fixed (?)

For me the idea behind RC is to limit the audience for the "first wave" by people who are ready to participate in testing and sharing their feedback. Those people will be confused less if there are some issues and will report with higher chances - they are enthusiasts and they know what they are doing.

If we release the RTM only, it could be potentially installed by wider range of people. They might become confused if there are some basic errors and might decide to postpone usage of product till some non-determined point of time in future.

Another story is SemVer. If it happens that we need to apply breaking changes to fix something (e.g. with NSubstitute integration) we'll be still enabled to apply them. On other hand, if we release `v4`, we'll need to release `v5` to fix them or work around with obsoletes.

Given that RC costs nothing, I don't see any issues with it. Do you see? I really want people to test .NET Core support and share the feedback as I'm not 100% sure we don't have some dummy issues with that.

@moodmosaic Have I convinced you? 😄

**Comment by moodmosaic** [comment]
I don't know... I'd rather go straight and release version 4.0.0... Unless @AutoFixture/core has idea(s), I'll leave the final decision to you.

**Comment by zvirja** [comment]
@moodmosaic Thanks for sharing your opinion! 👍 Could you also describe whether you see any issues with that?

Well, we have some time before the release, so let's see what @adamchester and @ecampidoglio think.

**Comment by moodmosaic** [comment]
>Could you also describe whether you see any issues with that?

With what? Avoiding `RC` and `RTM` is, well, just my opinion 😉 It doesn't mean you *have* to do it this way...

**Comment by Kralizek** [comment]
Just to clarify, @zvirja, are you considering having a pre-release published on NuGet?

**Comment by zvirja** [comment]
@Kralizek Yep, I have such a plan, unless @adamchester and @ecampidoglio (if he still participates in the project) have strong concerns against that.

**Comment by zvirja** [comment]
@adamchester We are very close to the v4 RC release. Currently all the main work has been done and I'm waiting till all the PRs are merged to proceed with the final preparations. It would be awesome if you could allocate an hour of free time and review the pending PRs. That will allow to close them quicker and all we will not to wait for 1 week for their expiration.

Hope you are capable of allocating that time and we'll proceed quicker😊

**Comment by zvirja** [comment]
I've finished all the release preparations.

@moodmosaic Could you please review the Release Notes on the wiki and announcement in Readme? Here are the links:
ReadMe: https://github.com/AutoFixture/AutoFixture/tree/v4
Wiki: https://github.com/AutoFixture/AutoFixture/wiki/v4.0-Release-Notes

It could happen that I made some stupid typos, so fresh look might help 😉 Thanks in advance.

**Comment by zvirja** [comment]
Huh, finally done! v4 RC1 has been released a few moments ago! 🎉🎈 ✨🌟 

Would like to thank everyone who participated in this release. This was quite a long way and we spent huge amount of resources to make this happen.

@moodmosaic Thanks for your active participation in the code review and various discussions. I know that sometimes you were overwhelmed by amount of the required attention, but you always found free time to follow up! This release wouldn't happen without your investments. 🍻🏇🥇 

@adamchester @ecampidoglio Thanks for your cooperation and help with code review and processes establishment. Unfortunately, I didn't have a chance to closely collaborate with you as you were busy most of the time, however I hope that you'll get more free in future and will continue to participate.

@ploeh Would like to thank you for the great product you made 🤝 Also thank you for all the collaboration with ownership transfer - it went smoothly as you always assisted me when needed.

Also I'd like to thank to everybody who participated in various discussions, actively followed the project, fired PRs and collaborated there. Here are some names, but I could easily miss somebody: @Kralizek @frblondin @sergeyshushlyapin @steve-codemunkies @adamralph @blairconrad. Thank you guys 🙏🍺 

I was happy to participate in this process and feel that we are making something awesome. Thanks for the great opportunity 😊
